openjdk.test.jck/src/test.jck/net/adoptopenjdk/stf/Jck.java

```diff
@@ -0,0 +1,968 @@
+		String kdcRealmName = null;
+		String kdcHostName = null;
+		BufferedReader br = new BufferedReader(new FileReader(krbConfFile.toString()));
+		String thisLine = null;
+	
+		while ((thisLine = br.readLine()) != null ) {
+			if (thisLine.contains("default_realm")) {
+				String[] parts = thisLine.split("=");
+				kdcRealmName = parts[1].trim();
+			}
+			if (thisLine.contains("admin_server")) {
+				String[] parts = thisLine.split("=");
+				kdcHostName = parts[1].trim();
+			}
+		 }
+		 br.close();
+		 return new KerberosConfig(kdcRealmName, kdcHostName);
+	}
+
+	private int getFreePort() throws Exception {
+		int freePort = -1;
+		
+		ServerSocket s = new ServerSocket(0);
+		freePort = s.getLocalPort();
+		s.close();
+		
+		return freePort;
+	}
+	
+	private String locateFileOrFolder(String root, String pattern) throws Exception {
+		String matches = "";
+		
+		StringBuffer sb = new StringBuffer();
+		File f = new File(root);
+		File[] files = f.listFiles();
+		for (File file : files) {
+			if (file.getName().contains(pattern)) {
+				sb.append(file.toString()).append(File.pathSeparator);
+			}
+			
+			if (file.isDirectory() && !(file.getName().contains("ext")) ) {
+				String s = locateFileOrFolder(file.toString(), pattern);
+				if (!s.equals("")) {
+					sb.append(s);
+				}
+			}		
+		}
+		matches = sb.toString();
+		return matches;
+	}
+	
+	private String getSignatureTestJars(String root) throws Exception {
+		String jarsList = "";
+		StringBuffer sb = new StringBuffer();
+		
+		sb.append(locateFileOrFolder(root,".jar"));
+		
+		String newRoot = root.replaceAll("lib", "bin");
+		sb.append(locateFileOrFolder(newRoot,"vm.jar"));
+		
+		jarsList = sb.toString();
+		return jarsList;
+	}
+		
+	public void jck8ConfigurationForMultijvmWithNoAgent(StfCoreExtension test) throws Exception {
+
+		String pathToJava = test.createJavaProcessDefinition().getCommand();
+		String pathToRmic = testJdk + File.separator + "bin" + File.separator + "rmic";
+		String pathToLib = testJdk + File.separator + "jre" + File.separator + "lib";
+		String pathToJavac = testJdk + File.separator + "bin" + File.separator + "javac";
+		DirectoryRef jckRuntimeNativeLibValue = nativesLoc;
+		DirectoryRef jckRuntimeJmxLibValue = nativesLoc;
+		String concurrency = "";
+		String keyword = "";
+		String libPath = "";
+		String testPlatform = "";
+		String robotAvailable = "";
+		String hostname = "";
+		String ipAddress = "";
+		extraJvmOptions = "";
+		
+		extraJvmOptions += " " + test.getJavaArgs(test.env().primaryJvm()) + " ";
+
+		InetAddress addr = InetAddress.getLocalHost();
+		ipAddress = addr.getHostAddress();
+		hostname = addr.getHostName();
+
+		freePort = getFreePort();
+		
+		if (freePort == -1) {
+			throw new StfException("Unable to get a free port");
+		}
+		
+		// Runtime settings
+		if (testSuite == TestSuite.RUNTIME) {
+			keyword = "keywords !interactive";
+			
+			if (platform.equals("win32")) {
+					testPlatform = "Windows";
```

Comment:
Could we fix the indentation in this if-else block below?

Reply:
Done